Tools/WinMLRunner/src/Run.cpp

```diff
@@ -6,13 +6,118 @@
+		}
+		m_length = value;
+	}
+
+	HRESULT __stdcall Buffer(uint8_t** value) noexcept final
+	{
+		*value = m_buffer;
+		return S_OK;
+	}
+
+private:
+	uint8_t* const m_buffer{};
+	uint32_t const m_capacity;
+	uint32_t m_length{};
+};
+
+LearningModel LoadModelCryptography(const std::wstring path)
+{
+	LearningModel model = nullptr;
+	try
+	{
+		// Making a custom buffer of the model size
+		std::ifstream in(path.c_str(), std::ifstream::ate | std::ifstream::binary);
+		std::ifstream::pos_type filesize = in.tellg();
+		std::vector<uint8_t> storage(filesize);
+		IBuffer ModelBuffer = make<CustomBuffer>(storage.data(), storage.size());
+
+		// Loading into storage file and getting the ONNX model as stream
+		StorageFile file = StorageFile::GetFileFromPathAsync(path.c_str()).get();
+		IInputStream stream = file.OpenAsync(FileAccessMode::Read).get();
+
+		//Reading the stream into an Ibuffer
+		stream.ReadAsync(ModelBuffer, ModelBuffer.Capacity(), InputStreamOptions::None).get();
+
+		//Encrypt the model using symmetric algorithm 
+		//PKCS7 algorithm used which automatically pads message to an appropriate length 
+		SymmetricKeyAlgorithmProvider alg = SymmetricKeyAlgorithmProvider::OpenAlgorithm(SymmetricAlgorithmNames::AesCbcPkcs7());
+		IBuffer symmbuff = winrt::Windows::Security::Cryptography::CryptographicBuffer::GenerateRandom(32);
+		CryptographicKey symmkey = alg.CreateSymmetricKey(symmbuff);
+		winrt::hstring halgName = alg.AlgorithmName();
+		std::string strAlgName = to_string(halgName);
+		IBuffer iv = NULL;
+		if (strAlgName.find("CBC") != std::string::npos)
+		{
+			iv = winrt::Windows::Security::Cryptography::CryptographicBuffer::GenerateRandom(alg.BlockLength());;
+		}
+		IBuffer EncryptBuff = CryptographicEngine::Encrypt(symmkey, ModelBuffer, iv);
+
+		//Decrypt the model and writing the model to a stream
+		IBuffer DecryptBuff = CryptographicEngine::Decrypt(symmkey, EncryptBuff, iv);
+		InMemoryRandomAccessStream MemoryStream;
+		winrt::Windows::Foundation::IAsyncOperationWithProgress<uint32_t, uint32_t> CCC = MemoryStream.WriteAsync(DecryptBuff);
+		auto Randomstream = MemoryStream.as<IRandomAccessStream>();
+		RandomAccessStreamReference StreamRef = RandomAccessStreamReference::CreateFromStream(Randomstream);
+
+		//Loading the model from the stream
+		winrt::Windows::Foundation::IAsyncOperation<LearningModel> modelObject = LearningModel::LoadFromStreamAsync(StreamRef);
+		LearningModel DecryptedModel = modelObject.get();
+		model = DecryptedModel;
+	}
+	catch (hresult_error hr)
+	{
+		std::wcout << "Load Model: " << path << " [FAILED]" << std::endl;
+		std::wcout << hr.message().c_str() << std::endl;
+		throw;
+	}
+
+	return model;
+}
 
 LearningModel LoadModel(const std::wstring path, bool capturePerf, OutputHelper& output, const CommandLineArgs& args, uint32_t iterationNum, Profiler<WINML_MODEL_TEST_PERF>& profiler)
 {
     LearningModel model = nullptr;
     output.PrintLoadingInfo(path);
+	model = LoadModelCryptography(path);
```

Comment:
Why do we need this for saving output tensor results? Also, won't the below line of code:

``` model = LearningModel::LoadFromFilePath(path); ```

overwrite the loading of the model here?

Reply:
Have made correction in the latest commit.